src/shapes/Text/StyledText.ts

```diff
@@ -266,18 +264,11 @@ export abstract class StyledText<
    * @return {Object} style object
    */
   getCompleteStyleDeclaration(lineIndex: number, charIndex: number) {
-    const style = this._getStyleDeclaration(lineIndex, charIndex) || {},
-      styleObject: TextStyleDeclaration = {},
-      styleProps = (this.constructor as typeof StyledText)._styleProperties;
-    for (let i = 0; i < styleProps.length; i++) {
-      const prop = styleProps[i];
-      // @ts-expect-error TS complains even when we serve everything on a silver plate.
-      styleObject[prop] =
-        typeof style[prop] === 'undefined'
-          ? this[prop as keyof this]
-          : style[prop];
-    }
-    return styleObject;
+    return {
+      // @ts-expect-error readonly
+      ...pick(this, (this.constructor as typeof StyledText)._styleProperties),
+      ...(this._getStyleDeclaration(lineIndex, charIndex) || {}),
```

Comment:
I don't think there is a need to pick the props from this object because it is defined with these props

Reply:
While this change is not wrong and is fine, tidier, we still need to validate the comment of that user that suggested verbose code over a loop in this part because of performance issues.
Just keep in mind that we may end up just putting a blunt list of properties in the style of:

```
return {
  fontsSize: style.fontSize ?? this.fontSize,
  ...
}
```

if performance concerns are true

Reply:
performance aside, I would also opt for the more human readable code

Reply:
Yes, as reported before, this function is a hot one as it's called **very** frequently when rendering, especially without caching enabled, so any performance issue is multiplied by a big factor.

In my findings with fabric v3.6, changing from a dynamically created object to a static one would improve the performance significantly:

## Before

![Screenshot 2023-03-06 at 12 06 07](https://github.com/fabricjs/fabric.js/assets/10067273/7224a25b-e7f2-4f0d-9462-7f0f7c5a158e)

## After

![Screenshot 2023-03-06 at 12 34 00](https://github.com/fabricjs/fabric.js/assets/10067273/08b4c88e-e3ff-4a78-a836-24a657d03be8)

Looping through the list of properties has a cost, but more significantly, creating an object by dynamically assigning the properties has a much greater impact I reckon. It means that each time the method is called, V8 or any other JS engine cannot optimize this code as it's too dynamic. Using an object with a static shape allows the engine to optimize the code and speed it up: https://mathiasbynens.be/notes/shapes-ics#shapes

Reply:
So for the sake of discussion
```ts
{
   fontSize: this.fontSize,
   ...,
   ...(this._getStyleDeclaration(lineIndex, charIndex) || {})
}
```

Is not good but what @asturur suggested is?

Reply:
I found today this tool that can maybe solve any doubt about what works and what doesn't: https://mathiasbynens.be/notes/shapes-ics#shapes